### ISSUE TITLE
[r] Assess and improve code coverage

### ISSUE THREAD

**Issue by atolopko-czi** [single-cell-data/TileDB-SOMA#842]
[Split off from multi-language test coverage story: #778]

- [ ] Assess code coverage for R code base. If deemed acceptable, close this issue.
- [ ] If coverage can be improved, add necessary tests or enumerate in new issue(s) the areas where coverage can be reasonably improved. 

Helpful link: https://app.codecov.io/gh/single-cell-data/TileDB-SOMA/

Blocked on #1023.
